src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/AddDefendantLitigationFriendCallbackHandler.java

```diff
@@ -48,16 +58,43 @@
     }
 
     private CallbackResponse aboutToSubmit(CallbackParams callbackParams) {
-        CaseData caseDataUpdated = callbackParams.getCaseData().toBuilder()
-            .businessProcess(BusinessProcess.ready(ADD_DEFENDANT_LITIGATION_FRIEND))
-            .respondent1LitigationFriendDate(LocalDateTime.now())
-            .respondent1LitigationFriendCreatedDate(
-                ofNullable(callbackParams.getCaseData().getRespondent1LitigationFriendCreatedDate())
-                    .orElse(LocalDateTime.now()))
-            .build();
+        CaseData caseData = callbackParams.getCaseData();
+        LocalDateTime currentDateTime = time.now();
+
+        CaseData.CaseDataBuilder caseDataUpdated = caseData.toBuilder()
+            .businessProcess(BusinessProcess.ready(ADD_DEFENDANT_LITIGATION_FRIEND));
+
+        if (caseData.getSelectLitigationFriend() == null) {
+            caseDataUpdated
+                .respondent1LitigationFriendDate(currentDateTime)
+                .respondent1LitigationFriendCreatedDate(
+                    ofNullable(callbackParams.getCaseData().getRespondent1LitigationFriendCreatedDate())
+                        .orElse(currentDateTime));
+        } else if (caseData.getSelectLitigationFriend().getValue().getLabel() != "Both") {
+            if (caseData.getSelectLitigationFriend().getValue().getLabel().contains("Respondent Two")
```

Comment:
This logic doesnt seem correct. If they select respondent 1 from the list. 
We want to build `respondent1LitigationFriend` only, i.e. Add Date + Created Date

If they select respondent 2 from the list, we want to 
We want to build respondent2LitigationFriend only - this will consist of copying the details entered from `respondent1LitigationFriend` field into `respondent2LitigationFriend` 
+ add the date and created date for respondent 2 field
+ then Nullifying respondent one

But this is a bit risky.. preferably, we should have a generic capture form for litigation friend that we can always clear down. The logic you build will copy the contents from the generic form, into respondent1LitigationFriend or respondent2LitigationFriend, or Both

What are your thoughts on this @akriti-hmcts ?